Tests/SwiftProtobufTests/Test_Enum.swift

```diff
@@ -40,6 +41,23 @@ class Test_Enum: XCTestCase, PBTestHelpers {
         }
     }
 
+    func testJSONdecodingOptions() {
+
+        var options = JSONDecodingOptions()
+        options.ignoreUnknownFields = true
+
+        assertJSONDecodeSucceeds("{\"optionalNestedEnum\":\"NEW_VALUE\"}", options: options) { (m: MessageTestType) -> Bool in
+            switch m.optionalNestedEnum {
+            case .zero: return true
+            default: return false
+            }
+        }
+
+        assertJSONDecodeSucceeds("{\"repeatedNestedEnum\":[\"FOO\",\"NEW_VALUE\"]}", options: options) { (m: MessageTestType) -> Bool in
+            m.repeatedNestedEnum == [.foo]
+        }
+    }
+
```

Comment:
we'll need to check `map<*, Enum>`, and when the single field is within a `oneof`.